test/secmemtest.c

```diff
@@ -90,6 +96,11 @@ int main(int argc, char **argv)
         perror_line();
         return 1;
     }
+    /* Test for infinite loop bug in sh_init, should fail */
+    if (CRYPTO_secure_malloc_init(1 << 30, 1)) {
```

Comment:
Why do you think this will fail?  Sure, on a machine with 32-bit `size_t`, but on a machine with 64-bit, you're nog even coming near `INT_MAX`.  I suggest this instead:

``` C
    if (CRYPTO_secure_malloc_init(1 << (sizeof(size_t) * 8 - 2), 1)) {
```